dev/hal/src/Buzzer.cpp

```diff
@@ -8,27 +8,39 @@
  * @copyright Copyright (c) 2024
  *
  */
-#include "mcal/inc/stm32f103xx.h"
+#include "stm32f103xx.h"
 #include "Types.h"
 #include "Define.h"
 #include "Constant.h"
-#include "mcal/inc/Pin.h"
-#include "mcal/inc/Gpio.h"
-#include "hal/inc/Buzzer.h"
+#include "Assert.h"
+#include "Rcc.h"
+#include "Pin.h"
+#include "Gpio.h"
+#include "Buzzer.h"
 
 using namespace stm32;
 using namespace stm32::dev::mcal::pin;
 using namespace stm32::dev::mcal::gpio;
 using namespace stm32::dev::hal::buzzer;
-
+using namespace stm32::dev::mcal::rcc;
 using namespace stm32::type;
 
 template<ConnectionType CT>
 Buzzer<CT>::Buzzer(const Pin& buzzerPin)
 : buzzerPin_(buzzerPin) {
+    STM32_ASSERT(buzzerPin.IsOutput());
+    InitializePins();
+}
+template<ConnectionType CT>
+void Buzzer<CT>::InitializePins() {
+    switch (buzzerPin_.GetPort()) {
+        case kPortA: Rcc::Enable(Peripheral::kIOPA); break;
+        case kPortB: Rcc::Enable(Peripheral::kIOPB); break;
+        case kPortC: Rcc::Enable(Peripheral::kIOPC); break;
+        default:STM32_ASSERT(false);                 break;
```

Comment:
It's ok but there is  a function that map `port` to `Peripheral` in util.
`MapPortToPeripheral()` to convert from `port` to `Peripheral`